Monika After Story/game/script-compliments.rpy

```diff
@@ -888,6 +889,174 @@ label mas_compliment_humor_3:
+            _("I was lonely waiting for you!"),
+            _("I've been thinking of you while you were gone!"),
+        ]
+        
+        missed_quips_upset_cynical = [
+            "...did you really even think of me at all?",
+            "...were you really thinking about me even a little?",
+            "That didn't feel very sincere, [player].",
+            "I'm not sure if you really mean that, [player].",
+            "If you really mean that, then please spend some time with me today.",
+            "If you really meant that, then you'd spend more time with me when you're here.",
+            "You say that, but I don't think you mean it one bit, [player].",
+            "You say that, but I feel like you don't mean it at all, [player]."
+        ]
+
+        missed_quips_upset_hope = [
+            "Thank you for showing me you still care, [player].",
+            "It means a lot to me that you were thinking of me.",
+            "I'm really glad to hear that, [player].",
+            "That's really nice to hear, coming from you.",
+            "I'm happy you've been thinking of me, [player].",
+            "[player]... I'm glad I'm not the only one.",
+            "That means the world to me, [player].",
+            "That makes me feel a lot better, [player]."
+        ]
+        
+        missed_quips_dis = [
+            "I'm not sure you mean that, [player]...",
+            "I doubt you mean that, [player]...",
+            "I don't think you really mean that, [player]...",
+            "If only you really meant that, [player]...",
+            "...Why do I think you just don't mean it?",
+            "...Why do I think you're just saying that?",
+            "...I can't really believe in that, [player].",
+            "I don't think that's true, [player]."
+        ]
+    
+    $ absence_length = mas_getAbsenceLength()
+    $ mas_flagEVL("mas_compliment_missed21", "CMP", EV_FLAG_HFM)
+    
+    if mas_isMoniNormal(higher=True):
+        $ mas_gainAffection(2)
+        $ hugchance = 2
+
+        #Really short absences
+        if absence_length < datetime.timedelta(hours=1):
+            $ hugchance = 0
+
+        if absence_length >= datetime.timedelta(days=3):
+            $ missedchance = renpy.random.randint(1,3)
+            if missedchance == 1:
+                m 6wub "I missed you a ton, [player]!"
+            elif missedchance == 2:
+                m 6wub "I missed you a whole lot, [player]!"
+            else:
+                m 6wubfb "...I ree{w=0.1}ee{w=0.1}ee{w=0.1}ally missed you [player]!"
+            $ hugchance = 17
+            if absence_length >= datetime.timedelta(days=7):
+                $ hugchance = 50
+        else:
+            m 6eub "I missed you too, [player]!"
+
+        $ missed_quip = random.choice(missed_quips)
+        show monika 2hublb
+        $ renpy.say(m, missed_quip)
+        if mas_isMoniEnamored(higher=True) and renpy.random.randint(1,50) <= hugchance:
```

Comment:
We should check when was the last hug.
```suggestion
        if (
            mas_isMoniEnamored(higher=True)
            and mas_timePastSince(persistent._mas_last_hold_dt, datetime.timedelta(hours=12))
            and renpy.random.randint(1, 50) <= hugchance
        ):
```

Reply:
Resolved in https://github.com/Monika-After-Story/MonikaModDev/pull/7419/commits/95507c54619f07b5712e9c8768214c151988f092.